streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -318,14 +314,7 @@ Map<MetricName, Metric> producerMetrics() {
     }
 
     Set<String> producerClientIds() {
```

Comment:
Could we use `String` instead of `Set<String>>` for now?

Reply:
Yes, but not "completely". The `ThreadMetadata` interface is public API, and thus it's impl cannot be changed.

Would need a KIP to change it -- not sure if worth doing at this point thought...